File: dev/run_tests.pl
my $infolder = "./testinput";
my $outfolder = "./testoutput";

# If the test folder does not exist, create it and populate it with music for testing
unless ( -e $infolder ) {
	die( "Error, $infolder does not exist. Please create it and stick some FLAC files in there for testing\n" );
}

my @flacfiles = <"$infolder/*.flac"> ;
if (scalar(@flacfiles) == 0) {
	die("No flac files found in $infolder. Please put some in there for testing (no subfolders please)...\n");
}

unless ( -e $outfolder ) { mkdir($outfolder); }

my @testypes = ("mp3","vorbis","flac","aacplusnero");

for $test (@testypes) {
	my $larg = "--lame-options='-preset standard' ";
	my $aarg = "--aacplus-options 64";
	my $varg = "--vorbis-options='quality=5:resample 32000:downmix'";

	for $opt ('-c','-f','-t 4','-n') {
		my $cmd = "python ./__main__.py $test $larg $aarg $varg $opt -o $outfolder/$test/ $infolder\n";
		die($cmd);
		print '-'x80;
		print "Executing: $cmd";
		print '-'x80;
		$rc = system($cmd);
		if ($rc != 0) {
			die("ERROR Executing command: \"$cmd\"\n");
		}
	}
}

rmdir($outfolder);

File: dev/run_tests.py
#!/usr/bin/python2
import os,shutil
from sys import exit
import subprocess as sp
from time import sleep

infolder = "testinput"
outfolder = "testoutput"

if not os.path.exists(infolder):
	print "Error, %s does not exist. Please create it and stick some FLAC files in there for testing" % infolder
	exit(1)

files = os.listdir(infolder)
flacfiles = filter(lambda x: x.endswith(".flac"), files)
if len(flacfiles) == 0:
	print "No flac files found in %s. Please put some in there for testing (no subfolders please)"
	exit(2)

if not os.path.exists(outfolder): os.mkdir(outfolder)

testypes = ["mp3,vorbis","mp3","vorbis","flac","aacplusnero"];

for test in testypes:
	sleep(10)
	args = [
	 "--lame-options='-preset standard' ",
	 "--aacplus-options 'br 64'",
	 "--vorbis-options='quality=5:resample 32000:downmix'",
	 "--opus-options='music:vbr'"
	 ]

	for opt in ('-c','-f','-t 4','-n'):
		cmd = "python2 ./__main__.py %s %s %s -o %s %s" % (test,' '.join(args),opt,outfolder,infolder)
		print '-'*80
		print "Executing: %s" % cmd
		print '-'*80
		rc = sp.call(cmd,shell=True)
		if (rc != 0) :
			print "ERROR Executing command: \"%s\"\n" % cmd
			exit(rc)
		
	
#print "All successful! Deleting output folder"
#shutil.rmtree(outfolder)
print "Done!"
